load.php

```diff
@@ -477,17 +477,37 @@ function perflab_run_module_activation_deactivation( $old_value, $value ) {
 				return;
 			}
 
-			$current_user = get_current_user_id();
-			$dismissed    = array_filter( explode( ',', (string) get_user_meta( $current_user, 'dismissed_wp_pointers', true ) ) );
-
-			if ( ! in_array( 'perflab-module-migration-pointer', $dismissed, true ) ) {
-				return;
+			$current_user       = wp_get_current_user();
+			$current_user_roles = array_shift( $current_user->roles );
+
+			$args = array(
```

Comment:
That seems like a potentially very slow query to me

Reply:
Thank you for addressing this. Yes, using the LIKE comparison method slows down the query due to the serialized storage of that meta value. Is there an alternative approach available to access that value? I'm eager to explore any potential options. Thank you!

Reply:
I don't think there's a faster way to query this using WordPress methods. IMO it's okay to use this, but we should definitely limit the results using the `number` argument, to avoid problems at scale. Unlikely, but if a site had 100s of administrators, we'd probably run into memory limit issues.

I think we could just limit to something like 20, I think it's unlikely for the vast majority of sites to have more than 20 administrators.

Reply:
The only alternative I see would be to run a completely custom query against `usermeta`. But then we'd still have to manually clear caches, and it becomes a bit more error-prone. I'm not sure there's a real risk here with a potentially slow query, as it would only happen explicitly on the module activation action, so it's probably overkill.

How about we only run the `WP_User_Query` if `! wp_is_large_user_count()`? That would at least avoid problems for sites with tons of users. For those, we could fall back to just dealing with the current user.

Reply:
Thank you for sharing your thoughts. I have updated the query approach based on the feedback received. The PR is now ready for review.